tests/test_http_request.py

```diff
@@ -406,6 +406,30 @@ def test_default_encoding_mixed_data(self):
         self.assertQueryEqual(r2.body, b'%C2%B5one=two&price%C2%A3=%C2%A3+100')
         self.assertEqual(r2.headers[b'Content-Type'], b'application/x-www-form-urlencoded')
 
+    def test_custom_nonutf8_bytes_data(self):
+        data = {b'\xbc one': b'two', b'price': b'\xbf 100'}
+        r2 = self.request_class("http://www.example.com", formdata=data)
+        self.assertEqual(r2.method, 'POST')
+        self.assertEqual(r2.encoding, 'utf-8')
+        self.assertQueryEqual(r2.body, b'price=%BF+100&%BC+one=two')
+        self.assertEqual(r2.headers[b'Content-Type'], b'application/x-www-form-urlencoded')
+
+    def test_custom_nonascii_utf8_bytes_data(self):
+        data = {b'\xc2 one': b'two', b'price': b'\xa3 100'}
```

Comment:
I'm not sure how do these tests differ, as as far as I can see `b'\xbc'`, `b'\xbf'`, `b'\xc2'` and `b'\xa3` are all non-UTF8.

Reply:
I think they are all UTF-8, but in any case your point remains.

Reply:
All UTF-8 single-byte characters are lower than \x80 :) `b'\xc2\xa3'` is a single UTF-8 character.

Reply:
Sorry, I confused UTF-8 with Unicode :facepalm: